5.5/contrib/my.cnf.template

```diff
@@ -0,0 +1,21 @@
+[mysqld]
+user = mysql
+
+datadir = ${MYSQL_DATADIR}
+basedir = /opt/rh/mysql55/root/usr
+plugin-dir = /opt/rh/mysql55/root/usr/lib64/mysql/plugin
+
+general_log = ON
+
+# Disable unix socket
+socket =
+
+# Disabling symbolic-links is recommended to prevent assorted security risks
+symbolic-links = 0
+
+# http://www.percona.com/blog/2008/05/31/dns-achilles-heel-mysql-installation/
+skip_name_resolve
+
+# Settings configured by the user:
+max_connections = ${MYSQL_MAX_CONNECTIONS}
```

Comment:
envsubst doesn't handle defaulting unfortunately.  otherwise this whole thing could be much cleaner.